core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1154,8 +1154,9 @@ class Partition(val topicPartition: TopicPartition,
 
           // Avoid writing to leader if there are not enough insync replicas to make it safe
           if (inSyncSize < minIsr && requiredAcks == -1) {
-            throw new NotEnoughReplicasException(s"The size of the current ISR ${partitionState.isr} " +
-              s"is insufficient to satisfy the min.isr requirement of $minIsr for partition $topicPartition")
+            throw new NotEnoughReplicasException(s"The size of the current ISR : $inSyncSize " +
```

Comment:
I agree the message is misleading. Thanks for fixing. Do we need to report the size at all? Could we just report the current ISR?
```scala
s"The current ISR $inSyncReplicaIds for topic partition $topicPartition does not have enough replicas to satisfy min.insync.replicas=$minIsr"
```
Or something like that?

Reply:
Hello @hachikuji , thank you for chiming in

While reviewing that part of the code history I understand the count was added on purpose, and not as a mistake done doing refactoring etc, that is why I kept it

I'm not sure it's useful, from my POV it would be best from a troubleshooting perspective to know which broker is out of ISR, but I wanted to keep that contribution simple :)